diff --git a/scripts/benchmark/run_dsae.sh b/scripts/benchmark/run_dsae.sh
deleted file mode 100755
index 758f912..0000000
--- a/scripts/benchmark/run_dsae.sh
+++ /dev/null
@@ -1,29 +0,0 @@
-python train.py -m \
-    hydra/launcher=joblib \
-    train.random_seed=831,277 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    train.random_seed=736,8 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    train.random_seed=172,828 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
\ No newline at end of file
diff --git a/scripts/benchmark/run_freeze.sh b/scripts/benchmark/run_freeze.sh
deleted file mode 100755
index 4397e43..0000000
--- a/scripts/benchmark/run_freeze.sh
+++ /dev/null
@@ -1,32 +0,0 @@
-python train.py -m \
-    hydra/launcher=joblib \
-    model=FreezeDsae \
-    train.random_seed=831,277 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=FreezeDsae \
-    train.random_seed=736,8 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=FreezeDsae \
-    train.random_seed=172,828 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
\ No newline at end of file
diff --git a/scripts/benchmark/run_tsdsae.sh b/scripts/benchmark/run_tsdsae.sh
deleted file mode 100755
index f836098..0000000
--- a/scripts/benchmark/run_tsdsae.sh
+++ /dev/null
@@ -1,65 +0,0 @@
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=831 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=736 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=172 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=277 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=8 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=828 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
\ No newline at end of file
diff --git a/scripts/benchmark/run_tsdsae_woReg.sh b/scripts/benchmark/run_tsdsae_woReg.sh
deleted file mode 100755
index a32f070..0000000
--- a/scripts/benchmark/run_tsdsae_woReg.sh
+++ /dev/null
@@ -1,71 +0,0 @@
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=831 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    model.reg_weights=[0,0,0,0] \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=736 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    model.reg_weights=[0,0,0,0] \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=172 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    model.reg_weights=[0,0,0,0] \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=277 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    model.reg_weights=[0,0,0,0] \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=8 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    model.reg_weights=[0,0,0,0] \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
-
-python train.py -m \
-    hydra/launcher=joblib \
-    model=TsDsae \
-    train.random_seed=828 \
-    model.z_dim=8,16,32 \
-    optim.optimizer.amsgrad=True,False \
-    model.reg_weights=[0,0,0,0] \
-    logging.media_log.reconstruction=False \
-    logging.media_log.generation=False \
-    logging.media_log.z_swap=False \
-    logging.media_log.v_project=False \
\ No newline at end of file
diff --git a/src/models/base.py b/src/models/base.py
index 76d832e..39eeb8d 100644
--- a/src/models/base.py
+++ b/src/models/base.py
@@ -18,6 +18,7 @@ import pytorch_lightning as pl
 import src.metric as metric
 from src.models.modules.audio_synth import GflFromMel
 from src.utils.seq import pad_batch
+from src.utils.util import ensure_dir
 
 
 logger = logging.getLogger(__name__)
@@ -339,10 +340,13 @@ class DsaeBase(pl.LightningModule, ABC):
             'swap_n_clfr': score,
         }
         print(output_dict)
+        dataset = self.hparams.data.datasets.train._target_.split('.')[-1]
         orig_cwd = Path(hydra.utils.get_original_cwd())
         fname = "_".join([f"{k}={v}" for k, v in output_dict['params'].items()])
-        fname = orig_cwd / 'benchmark' / f'{run_id}_{fname}.json'
-        with open(fname, 'w', encoding='utf-8') as f:
+        fname = f'{run_id}_{fname}.json'
+        tgt_dir = orig_cwd / 'benchmark' / dataset
+        ensure_dir(tgt_dir)
+        with open(tgt_dir / fname, 'w', encoding='utf-8') as f:
             json.dump(output_dict, f, ensure_ascii=False, indent=4)
 
     def _run_step(self, batch, prefix):
